modules/programs/ghostty.nix

```diff
@@ -114,6 +114,23 @@ in
         defaultText = lib.literalMD "`true` if programs.ghostty.package is not null";
       };
 
+      systemd = lib.mkOption {
+        type = lib.types.submodule {
+          options = {
+            enable = lib.mkEnableOption "the Ghostty systemd user service" // {
+              description = "Whether to enable the systemd user service for Ghostty.";
```

Comment:
Mostly looks good to me now, I just think we should probably enable this by default on Linux. Also instead of enabling it conditionally on Linux, we should probably just add an assertion for it. Also there's a typo on the commit message

Reply:
I've dropped the assertion and instead make all of this conditional based on if `cfg.package` is set

Reply:
> I've dropped the assertion and instead make all of this conditional based on if cfg.package is set

We want assertions for both the package and the platform